diff --git a/net/netmon/netmon_polling.go b/net/netmon/netmon_polling.go
--- a/net/netmon/netmon_polling.go
+++ b/net/netmon/netmon_polling.go
@@ -1,6 +1,6 @@
 // Copyright (c) Tailscale Inc & contributors
 // SPDX-License-Identifier: BSD-3-Clause
 
-//go:build (!linux && !freebsd && !windows && !darwin) || android
+//go:build ((!linux && !freebsd && !windows && !darwin) || android) && !ts_include_cli
 
 package netmon
